backend/hdf5/h5x/H5DataType.cpp

```diff
@@ -281,6 +281,12 @@ data_type_from_h5(H5T_class_t vclass, size_t vsize, H5T_sign_t vsign)
     return DataType::Nothing;
 }
 
+void check_bool_enum(h5x::DataType type) {
+    assert(type.member_count() == 2);
```

Comment:
I used `assert` since it's also used in `data_type_from_h5` when the the `vclass` is unknown. It's also used when the member count in the compound data type is wrong.

Should those cases be throwing exceptions as well?